core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java

```diff
@@ -592,8 +593,11 @@ public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredComm
          } else if (trace) {
             log.trace("Cannot remove entry as its lifespan or value do not match");
          }
-      } else if (trace) {
-         log.trace("Nothing to remove since the entry doesn't exist in the context or it is already removed");
+      } else {
+         if (trace) {
+            log.trace("Nothing to remove since the entry doesn't exist in the context or it is already removed - assume command was successful");
```

Comment:
We don't want to let something like an eviction to cause the expiration command to fail. If the value isn't present we have to assume it can be expired.